src/WpfMath/Controls/FormulaControl.cs

```diff
@@ -0,0 +1,223 @@
+    private static readonly TexFormulaParser _formulaParser = WpfTeXFormulaParser.Instance;
+    private TexFormula? texFormula;
+    private Size _bounds;
+
+    public string Formula
+    {
+        get { return (string)GetValue(FormulaProperty); }
+        set { SetValue(FormulaProperty, value); }
+    }
+
+    public double Scale
+    {
+        get { return (double)GetValue(ScaleProperty); }
+        set { SetValue(ScaleProperty, value); }
+    }
+
+    public string SystemTextFontName
+    {
+        get => (string)GetValue(SystemTextFontNameProperty);
+        set => SetValue(SystemTextFontNameProperty, value);
+    }
+
+    public bool HasError
+    {
+        get { return (bool)GetValue(HasErrorProperty); }
+        private set { SetValue(HasErrorProperty, value); }
+    }
+
+    public ObservableCollection<Exception> Errors
+    {
+        get { return (ObservableCollection<Exception>)GetValue(ErrorsProperty); }
+        private set { SetValue(ErrorsProperty, value); }
+    }
+
+    public ControlTemplate ErrorTemplate
+    {
+        get { return (ControlTemplate)GetValue(ErrorTemplateProperty); }
+        set { SetValue(ErrorTemplateProperty, value); }
+    }
+
+    public int SelectionStart
+    {
+        get { return (int)GetValue(SelectionStartProperty); }
+        set { SetValue(SelectionStartProperty, value); }
+    }
+
+    public int SelectionLength
+    {
+        get { return (int)GetValue(SelectionLengthProperty); }
+        set { SetValue(SelectionLengthProperty, value); }
+    }
+
+    public Brush? SelectionBrush
+    {
+        get => (Brush)GetValue(SelectionBrushProperty);
+        set => SetValue(SelectionBrushProperty, value);
+    }
+
+    public static readonly DependencyProperty FormulaProperty = DependencyProperty.Register(
+        nameof(Formula), typeof(string), typeof(FormulaControl),
+        new FrameworkPropertyMetadata(
+            "",
+            FrameworkPropertyMetadataOptions.AffectsMeasure,
+            null,
+            coerceValueCallback: CoerceFormula));
+
+    public static readonly DependencyProperty ScaleProperty = DependencyProperty.Register(
+        nameof(Scale), typeof(double), typeof(FormulaControl),
+        new FrameworkPropertyMetadata(
+            20d,
+            FrameworkPropertyMetadataOptions.AffectsMeasure,
+            null,
+            coerceValueCallback: CoerceScaleValue));
+
+    public static readonly DependencyProperty SystemTextFontNameProperty = DependencyProperty.Register(
+        nameof(SystemTextFontName), typeof(string), typeof(FormulaControl),
+        new FrameworkPropertyMetadata(
+            "Arial",
+            FrameworkPropertyMetadataOptions.AffectsMeasure));
+
+    public static readonly DependencyProperty SelectionStartProperty = DependencyProperty.Register(
+        nameof(SelectionStart), typeof(int), typeof(FormulaControl),
+        new FrameworkPropertyMetadata(
+            0,
+            FrameworkPropertyMetadataOptions.AffectsMeasure));
```

Comment:
Does this really change the size of the control? If not, AffectsRender should suffice. Same with SelectionLength and SelectionBrush.